### PR TITLE
Fix curb on debug

### PR DESCRIPTION
**Pull request by pavel-jurasek-bcgdv-com**
This PR fixing issue: https://github.com/bblimke/webmock/issues/621

### CODE REVIEW

**Comment by pavel-jurasek-bcgdv-com** [comment]
https://travis-ci.org/pavel-jurasek-bcgdv-com/webmock should work

**Comment by bblimke** [comment]
@pavel-jurasek-bcgdv-com thanks for that.

I'm wondering if "faking" on_debug actually makes sense.
This code just pretends to be a real Curl.

If someone wants to debug what's going on, I'm not sure seeing a data generated by WebMock,
is what user wants to see.

What would be the usecase of using on_debug with WebMock enabled?

**Comment by pavel-jurasek-bcgdv-com** [comment]
@bblimke well i add on_debug statements just because simplecov do not hit then all code.

Use case:
Building abstract business client (internally using curb). If this client will contain debug option (Foo.client.debug = true) and within the class will be specified:

```ruby
@curb.on_debug do |foo, bar|
  >> loging to NewRelict in my case
end
```
 test coverage will be not 100% since on_debug is not implemented in WebMock. It will produce 100% coverage without WebMock. I can simplify on_debug statements but i tried to produce as close as possible results.

**Comment by bblimke** [comment]
I see. Thanks for explaining.

What you implemented, is it exactly what curb would output every time?

I wouldn't want to mislead people who use on_debug to trace some issue, with incorrect data.

**Comment by pavel-jurasek-bcgdv-com** [comment]
@bblimke as close as i can go it not handle follow location requests but for intention (e.g. on_debug handling) should be fine.

File: lib/webmock/http_lib_adapters/curb_adapter.rb [hunk 1]
```diff
@@ -77,6 +77,8 @@ def curb_or_webmock
         else
           raise WebMock::NetConnectNotAllowedError.new(request_signature)
         end
+
+        build_webmock_response if defined?( @on_debug )
```
Comment (bblimke) [inline lib/webmock/http_lib_adapters/curb_adapter.rb]: why is that needed? isn't webmock response already built?

**Comment by pavel-jurasek-bcgdv-com** [comment]
@bblimke removed :) thanks to check code ! i over looked this.

**Comment by bblimke** [comment]
Excellent :)

**Comment by bblimke** [comment]
released as 2.1.0
